src/components/plan/PlanForm.svelte

```diff
@@ -153,19 +170,41 @@
     </fieldset>
     <fieldset>
       <Collapse title="Snapshots" padContent={false}>
-        <button class="st-button secondary" slot="right" on:click={onCreatePlanSnapshot}>Take Snapshot</button>
+        <div class="buttons" slot="right">
+          {#if $simulationDatasetId >= 0}
+            <FilterToggleButton
+              label="Snapshot"
+              offTooltipContent="Filter snapshots by selected simulation"
+              onTooltipContent="Remove filter"
+              isOn={isFilteredBySimulation}
+              on:toggle={onToggleFilter}
+            />
+          {/if}
+          <button class="st-button secondary" on:click={onCreatePlanSnapshot}>Take Snapshot</button>
+        </div>
         <div style="margin-top: 8px">
           <CardList>
-            {#each $planSnapshotsWithSimulations as planSnapshot (planSnapshot.snapshot_id)}
+            {#each filteredPlanSnapshots as planSnapshot (planSnapshot.snapshot_id)}
               <PlanSnapshot
                 activePlanSnapshotId={$planSnapshotId}
                 {planSnapshot}
-                on:click={() => setQueryParam(SearchParameters.SNAPSHOT_ID, `${planSnapshot.snapshot_id}`, 'PUSH')}
+                on:click={() => {
+                  setQueryParam(SearchParameters.SNAPSHOT_ID, `${planSnapshot.snapshot_id}`, 'PUSH');
+                  $planSnapshotId = planSnapshot.snapshot_id;
+
+                  if (planSnapshot.simulation?.id != null) {
+                    setQueryParam(SearchParameters.SIMULATION_DATASET_ID, `${planSnapshot.simulation?.id}`, 'PUSH');
+                    $simulationDatasetId = planSnapshot.simulation?.id;
+                  } else {
+                    removeQueryParam(SearchParameters.SIMULATION_DATASET_ID);
+                    $simulationDatasetId = -1;
```

Comment:
Setting this to -1 does not appear to always clear simulation results (or perhaps this comment is related to a different section of code?). Seeing an issue if you load a snapshot with a completed sim (which loads the sim dataset) and then click on a snapshot that does not have any associated sims. Result is that the sim data will not clear though if you refresh the page the stale sim data won't be there.

Reply:
what do you mean by the sim data? I think i'm not seeing where this data is being displayed

Reply:
Sim data meaning the resulting spans that should be shown in the UI (timeline, table, etc).

Reply:
Thanks! I see it now